### PR TITLE
Add a commande line option to osrm-routed for max locations supported in distance table query

### PR DESCRIPTION
**Pull request by frodrigo**
Due to issue (with gcc or luabind depends on computer) to compile develop branch this patch is not tested on this branch. Nevertheless this patch work on last release. #1236

### CODE REVIEW

**Comment by TheMarex** [comment]
@frodrigo Added some comments to the source. Once this is fixed this should be good to merge. :-)

**Comment by frodrigo** [comment]
I did what you asked

**Comment by frodrigo** [comment]
Fix again with stash

**Comment by TheMarex** [comment]
Looks good! Once the build + tests pass I will merge this. Thanks again @frodrigo!

File: Include/osrm/ServerConfig.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/*
+
+Copyright (c) 2014, Project OSRM, Dennis Luxen, others
```
Comment (DennisOSRM) [inline Include/osrm/ServerConfig.h]: 2015

File: Include/osrm/ServerConfig.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#include <osrm/ServerPaths.h>
+
+struct ServerConfig
+{
+    ServerConfig()
+        : use_shared_memory(false)
+        , max_locations_distance_table(100)
+    {}
+
+    ServerConfig(const ServerPaths paths, const bool flag, const int max)
```
Comment (DennisOSRM) [inline Include/osrm/ServerConfig.h]: could we pass ServerPaths by ref?

File: Include/osrm/ServerConfig.h [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        : server_paths(paths)
+        , use_shared_memory(flag)
+        , max_locations_distance_table(max)
+    {}
+
+    ServerPaths server_paths;
+    bool use_shared_memory;
```
Comment (DennisOSRM) [inline Include/osrm/ServerConfig.h]: The order of members should be from larger to smaller, here.

File: Library/OSRM.h [hunk 4]
```diff
@@ -46,7 +46,7 @@ class OSRM
     std::unique_ptr<OSRM_impl> OSRM_pimpl_;
 
   public:
-    explicit OSRM(ServerPaths paths, const bool use_shared_memory = false);
+    explicit OSRM(ServerConfig serverConfig);
```
Comment (DennisOSRM) [inline Library/OSRM.h]: Can we pass the object by const ref?

**Comment by DennisOSRM** [comment]
Added some comments to the code.

@TheMarex hold with merging until the next release is tagged this week. The patch breaks the API that the node bindings use.

**Comment by TheMarex** [comment]
@DennisOSRM exactly, I need this option for node-osrm. So it would be best if we merge this pre-tag, right?

**Comment by DennisOSRM** [comment]
@TheMarex can it wait for a day or two?

File: Util/ProgramOptions.h [hunk 5]
```diff
@@ -272,6 +276,11 @@ inline unsigned GenerateServerProgramOptions(const int argc,
     {
         return INIT_OK_START_ENGINE;
     }
+    if (1 > max_locations_distance_table)
+    {
+        throw OSRMException("Max location for distance table must be a positive number");
```
Comment (DennisOSRM) [inline Util/ProgramOptions.h]: try using `osrm::exception`.

File: Include/osrm/ServerConfig.h (hunk 3 above)
Comment (frodrigo) [inline Include/osrm/ServerConfig.h]: Are you talking about size of the members name ? or the size in memory ?

Comment (DennisOSRM) [inline Include/osrm/ServerConfig.h]: Size in memory. ServerPaths >= int >= bool

File: Library/OSRM.h (hunk 4 above)
Comment (frodrigo) [inline Library/OSRM.h]: By ref yes, by const no

**Comment by frodrigo** [comment]
It's all right now ?

**Comment by DennisOSRM** [comment]
nope, the CI fail. As mentioned above, replace `OSRMException` with `osrm::exception`.

**Comment by frodrigo** [comment]
I use osrm::exception in last commit

**Comment by DennisOSRM** [comment]
triggered a rebuild of the PR

**Comment by DennisOSRM** [comment]
something is fishy with this PR. Did you rebase onto latest develop branch?

**Comment by DennisOSRM** [comment]
No idea what happened to this PR. Went ahead, merged manually into a new branch + PR: https://github.com/Project-OSRM/osrm-backend/pull/1335

Will handle it from there and merge once the CIs come back positive.

**Pull request closed without merging**